core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceGroupNetworkParameterConverter.java

```diff
@@ -0,0 +1,195 @@
+            List<String> subnetIds = key.getSubnetIds();
+            if (subnetIdsNotDefined(subnetIds)) {
+                response.setSubnetIds(subnetIds);
+            } else if (source.getValue() != null) {
+                response.setSubnetIds(List.of(source.getValue().getPreferedSubnetId()));
+            }
+        }
+
+        return response;
+    }
+
+    private InstanceGroupGcpNetworkV4Parameters convertToGcpNetworkParams(Pair<InstanceGroupGcpNetworkV1Parameters,
+            EnvironmentNetworkResponse> source) {
+        InstanceGroupGcpNetworkV1Parameters key = source.getKey();
+
+        InstanceGroupGcpNetworkV4Parameters response = new InstanceGroupGcpNetworkV4Parameters();
+
+        if (key != null) {
+            List<String> subnetIds = key.getSubnetIds();
+            if (subnetIdsNotDefined(subnetIds)) {
+                response.setSubnetIds(subnetIds);
+            } else if (source.getValue() != null) {
+                response.setSubnetIds(List.of(source.getValue().getPreferedSubnetId()));
+            }
+        }
+
+        return response;
+    }
+
+    private InstanceGroupYarnNetworkV4Parameters convertToYarnNetworkParams(Pair<InstanceGroupYarnNetworkV1Parameters,
+            EnvironmentNetworkResponse> source) {
+        return new InstanceGroupYarnNetworkV4Parameters();
+    }
+
+    private InstanceGroupAwsNetworkV4Parameters convertToAwsNetworkRequest(Pair<InstanceGroupAwsNetworkV1Parameters,
+            EnvironmentNetworkResponse> source) {
+        InstanceGroupAwsNetworkV1Parameters key = source.getKey();
+
+        InstanceGroupAwsNetworkV4Parameters response = new InstanceGroupAwsNetworkV4Parameters();
+
+        if (key != null) {
+            List<String> subnetIds = key.getSubnetIds();
+            if (subnetIdsNotDefined(subnetIds)) {
+                response.setSubnetIds(key.getSubnetIds());
+            } else if (source.getValue() != null) {
+                response.setSubnetIds(List.of(source.getValue().getPreferedSubnetId()));
+            }
+        }
+
+        return response;
+    }
+
+    private InstanceGroupOpenstackNetworkV4Parameters convertToOpenstackNetworkRequest(Pair<InstanceGroupOpenstackNetworkV1Parameters,
+            EnvironmentNetworkResponse> source) {
+        InstanceGroupOpenstackNetworkV1Parameters key = source.getKey();
+
+        InstanceGroupOpenstackNetworkV4Parameters response = new InstanceGroupOpenstackNetworkV4Parameters();
+
+        if (key != null) {
+            List<String> subnetIds = key.getSubnetIds();
+            if (subnetIdsNotDefined(subnetIds)) {
+                response.setSubnetIds(key.getSubnetIds());
+            } else if (source.getValue() != null) {
+                response.setSubnetIds(List.of(source.getValue().getPreferedSubnetId()));
+            }
+        }
+
+        return response;
+    }
+
+    private boolean subnetIdsNotDefined(List<String> subnetIds) {
+        return subnetIds != null && !subnetIds.isEmpty();
+    }
```

Comment:
The method name is confusing. It returns `true` if there are some subnet IDs in the list.